tests/lib/tools/log-analyzer

```diff
@@ -0,0 +1,511 @@
+        return self.lines[-1*self.lines_limit-1:]
+
+    def __repr__(self):
+        return ''.join(self._get_lines())
+
+    def __dict__(self):
+        return {'lines': self.lines[-1*self.lines_limit-1:]}
+
+
+class Operation:
+    """
+    Operation represents other actions that the spread running can do while executing tests like:
+    Rebooting, Discarding, Allocating, Waiting, Allocated, Connecting, Connected, Sending
+    """
+
+    def __init__(self, verb, task, extra, date, time, source_line):
+        self.type = OPERATION
+        self.verb = verb
+        self.time = time
+        self.extra = extra
+        self.date = date
+        self.task = task
+        self.source_line = source_line
+
+    def __repr__(self):
+        return self.source_line
+
+    def __dict__(self):
+        return {
+            'type': self.type,
+            'date': self.date,
+            'time': self.time,
+            'verb': self.verb,
+            'task': self.task,
+            'extra': self.extra
+            }
+
+
+class LogReader:
+    """
+    LogReader manages the spread log, it allows to read, export and print a spread log.
+    """
+    def __init__(self, filepath, output_type, lines_limit, store_setup):
+        self.filepath = filepath
+        self.output_type = output_type
+        self.lines_limit = lines_limit
+        self.store_setup = store_setup
+        self.lines = []
+        self.iter = 0
+        self.full_log = []
+
+    def __repr__(self):
+        return str(self.__dict__())
+
+    def __dict__(self):
+        return {'full_log': self.full_log}
+
+    def print_log(self, details, results):
+        if not self.full_log:
+            return
+        if details == ALL:
+            print(''.join(str(x) for x in self.full_log))
+        elif details == NONE:
+            pass
+        elif details == ERROR:
+            print(''.join(str(x) for x in self.full_log if x.type == INFO and \
+                x.info_type == ERROR_TYPE))
+        else:
+            print(''.join(str(x) for x in self.full_log if x.type == details))
+
+        if results == ALL:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT))
+        elif results == NONE:
+            pass
+        elif results == FAILED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == FAILED_TYPE))
+        elif results == ABORTED:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == ABORTED_TYPE))
+        else:
+            print(''.join(str(x) for x in self.full_log if x.type == RESULT and \
+                x.result_type == SUCCESSFUL_TYPE))
+
+
+    def export_log(self, filepath):
+        prepared_log = []
+        for item in self.full_log:
+            prepared_log.append(item.__dict__())
+        with open(filepath, 'w') as json_file:
+            json.dump(prepared_log, json_file)
+
+    def _next_line(self):
```

Comment:
I think you could just do `iter(self.ines)` and use the `next()` function to advance through this.